settings.py

```diff
@@ -0,0 +1,86 @@
+import pathlib
+from pathlib import Path
+from functools import lru_cache
+
+import sqlalchemy
+from pydantic import BaseSettings, PostgresDsn, Field, validator
+
+__all__ = ['BASE_DIR', 'ProjectSettings', 'TEMPLATES_PATH', 'STORAGE_PATH', 'SETTLEMENT_HOSTEL_PATH']
+
+BASE_DIR = pathlib.Path(__file__).resolve().parent
+TEMPLATES_PATH = Path(BASE_DIR) / "templates"
+STORAGE_PATH = Path(BASE_DIR) / "storage"
+SETTLEMENT_HOSTEL_PATH = Path(BASE_DIR) / (Path(STORAGE_PATH) / "settlement_hostel")
+
+
+class ProjectSettings(BaseSettings):
+    # DATABASE SETTINGS
+    POSTGRES_USER: str = Field(default='postgres')
+    POSTGRES_HOST: str = Field(default='localhost')
+    POSTGRES_PASSWORD: str = Field(default='postgres')
+    POSTGRES_PORT: int = Field(default=5432)
+    POSTGRES_DSN: PostgresDsn = Field(default=None)
+
+    # BACK-END SETTINGS
+    TOKEN_LIFE_TIME: str = Field(default="TOKEN_LIFE_TIME")
+    # 1 hour
+    ACCESS_TOKEN_EXPIRE_SECONDS: int = Field(default=3660)
+    # 7 days
+    REFRESH_TOKEN_EXPIRE_SECONDS: int = Field(default=(60 * 60 * 24 * 7))
+    ALGORITHM: str = Field(default="HS256")
+    DATETIME_FORMAT: str = Field("%Y-%m-%d %H:%M:%S")
+
+    # JWT SETTINGS
+    JWT_SECRET_KEY: str = Field('JWT_SECRET_KEY')
+    JWT_REFRESH_SECRET_KEY: str = Field('JWT_REFRESH_SECRET_KEY')
+
+    # CORE SETTINGS
+    CORS_ALLOW_CREDENTIALS: bool = Field(default=True)
+    CORS_ALLOW_HEADERS: list[str] = Field(default=["*"])
+    CORS_ALLOW_METHODS: list[str] = Field(default=["*"])
+    CORS_ALLOW_ORIGINS: list[str] = Field(default=["*"])
+
+    @validator("POSTGRES_DSN", always=True)
+    def get_postgres_dsn(cls, value: str, values: dict) -> 'URL' | str:
+        if isinstance(value, str):
```

Comment:
Note: according to type hint above `value: str`, this is always true. If `value` can belong to other types (e. g. `None` or `int`), then type hint must be updated to something like `value: Optional[str]` or `value: Union[str, int]` ([typing.Union](https://docs.python.org/3/library/typing.html#typing.Union)).

Though, you probably have to refactor this code portion completely according to @Kostiantyn-Salnykov's comment